### ISSUE TITLE
Use Composer instead of Git submodules?

### ISSUE THREAD

**Issue by hassankhan** [mdkholy/dashbrew#9]
Happy to create a PR if necessary

**Comment by mohatt** [comment]
Thanks @hassankhan for your suggestion. I have already considered using Composer for managing Dashbrew dependencies but currently I think Git submodules is doing just fine given the nature of the project and the minimal dependencies :). It will also make Dashbrew installation procedure simple by avoiding `composer install` command so that people can be up and running in no time.

**Comment by hassankhan** [comment]
Well you could always package Dashbrew into a PHAR. `composer install` can be done easily by any user, and you could set up the provisioner to run it surely?

**Comment by mohatt** [comment]
Well, using a PHAR may not be a good solution here because the dependencies are being used by both CLI (i.e. Dashbrew Provisioner) and web interfaces.
However, the idea of running `composer install` as part of the provisioning process sounds pretty good :+1:. I will make sure to implement it in the next release. Thank you  :-) .

**Comment by DomagojGojak** [comment]
Hi guys! I think that it would be better just to let users manually update the composer when they need to since that is a practice that they are already used to + vagrant provision will then maintain current time to do the provisioning. Another neat thing is that this would open a way to make dashbrew installable using the composer itself and then dashbrew could use the composer post update commands to do the provisioning which would lead to dashbrew installation by a single line in the terminal. Let me know what you guys think. Also, I would be happy to create the PR as well if needed. :-)

**Comment by mathewpeterson** [comment]
+1 to getting the dashbrew provisioner into a composer package. By doing this, we could more easily integrate it into our own toolchain.

**Comment by hassankhan** [comment]
Yep, could move all the Dashbrew PHP code to a new repo (dashbrew-core?) and have Composer install that onto the VM. If @mdkholy's cool with it, could someone cut a branch and we can maybe get some code up? Happy to do it myself
